src/services/VoteService.ts

```diff
@@ -36,4 +47,48 @@ export class VoteService {
       .map((option) => option.identifier)
       .includes(vote);
   }
+
+  public async countVotes(
+    ballotID: string
+  ): Promise<TotalVoteCount> {
+    if (await BallotRepository.getBallot(ballotID) == null)
+      throw new NotFound("Ballot with given id does not exist");
+
+    const votes = await this.getVotes(ballotID);
+    return {
+      count: votes.length
+    };
+  }
+
+  public async getVoteResult(
+    ballotID: string,
+    voteOptions: VotingOption[]
+  ): Promise<VoteResult[]> {
+    const votes = await this.getVotes(ballotID);
+
+    const results: Record<string, number> = {} as never
+
+    // init vote counter
+    for (const voteOption of voteOptions) {
+      results[voteOption.identifier] = 0
+    }
+
+    // count votes
+    for (const vote of votes) {
+      results[vote.vote]++;
+    }
+
+    // fill records into array list
+    const out: VoteResult[] = [];
+    for (const result of Object.getOwnPropertyNames(results)) {
+      const voteOption = voteOptions.find(vo => result == vo.identifier)
+      const vr: VoteResult = {
```

Comment:
```suggestion
      const voteResult = {
```

Reply:
Resolved in [#52 Renamed vr into voteResult](https://github.com/LeoTuet/backend-we-want-to-sleep/pull/54/commits/83b8907df1deead9c87cf87a9094963334be85ae)